python/templates/macros/collections.jinja2

```diff
@@ -206,7 +206,7 @@ podio::CollectionReadBuffers createBuffersV{{ schemaVersion }}(bool isSubset) {
 {% endfor %}
   }
 
-  readBuffers.createCollection = [](podio::CollectionReadBuffers buffers, bool isSubsetColl) {
+  readBuffers.createCollection = [](const podio::CollectionReadBuffers& buffers, bool isSubsetColl) {
```

Comment:
I am not particularly liking this because it's a bit misleading, because something down the line will be stealing the data from these buffers:
https://github.com/AIDASoft/podio/blob/53ded81210d556638e4c50a761bc501ec2adf12e/python/templates/CollectionData.cc.jinja2#L35-L36

On the other hand this is the case already now because the copy we make at the moment will also have the same behavior. In the end all of this should be cleaned up better at some point (see https://github.com/AIDASoft/podio/issues/506).

Interestingly this also seems to work even though we technically assign to a different `std::function`: https://github.com/AIDASoft/podio/blob/53ded81210d556638e4c50a761bc501ec2adf12e/include/podio/CollectionBuffers.h#L51